lib/src/core/models/group.dart

```diff
@@ -1,26 +1,34 @@
-import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:isar/isar.dart';
 import 'package:kana_to_kanji/src/core/constants/alphabets.dart';
 import 'package:kana_to_kanji/src/core/constants/kana_type.dart';
 
-part 'group.freezed.dart';
 part 'group.g.dart';
 
-@freezed
-class Group with _$Group {
-  const factory Group(
-      {required int id,
-      required Alphabets alphabet,
-      required String name,
-      // Type of Kana, only present when the alphabet is hiragana or katakana
-      KanaTypes? kanaType,
-      String? localizedName}) = _Group;
-
-  factory Group.fromJson(Map<String, Object?> json) => _$GroupFromJson(json);
-
-  static const tableCreate = "CREATE TABLE groups("
-      "   id INTEGER PRIMARY KEY AUTOINCREMENT,"
-      "   alphabet TEXT NOT NULL, "
-      "   name TEXT NOT NULL, "
-      "   localizedName TEXT, "
-      "   kanaType TEXT)";
+@collection
+@Name("Groups")
+class Group {
+  final int id;
+
+  @enumValue
+  final Alphabets alphabet;
+
+  final String name;
+
+  @enumValue
+  final KanaTypes kanaType;
+
+  final String? localizedName;
+
+  Group(this.id, this.alphabet, this.name, this.kanaType, this.localizedName);
+
+  factory Group.fromJson(Map<String, Object?> json) => Group(
+      json['id'] as int,
+      Alphabets.values
+          .where((element) => element.value == json['alphabet'])
+          .first,
+      json['name'] as String,
+      KanaTypes.values
+          .where((element) => element.value == json['kanaType'])
+          .first,
+      json['localizedName'] as String?);
```

Comment:
We have the [json_serializable](https://pub.dev/packages/json_serializable) package installed so no need to create the `fromJson`.

Add `@JsonSerializable()` as class annotation and replace the fromJson with `factory Group.fromJson(Map<String, dynamic> json) => _$Group.fromJson(json);` it should work 😄

Reply:
Done